rmd_files/reshaping.Rmd

```diff
@@ -4,411 +4,291 @@
 
-```{r, purl=purl_example_code, eval=T}
-#mapping all weeks to one variable called "weeks" 
-billboard %>% tidyr::pivot_longer(cols = starts_with("wk"), names_to = "weeks", values_to = "rank") %>% head()
+```{r, purl=purl_example_code ,eval=T}
+# adding a prefix to new columns
+wide_annual_offences <- annual_offences %>%
+  tidyr::pivot_wider(
+    names_from = 'year',
+    values_from = 'count',
+    names_prefix = 'count_'
+  )
+head(wide_annual_offences)
 ```
 
+When transforming count data like this we may have legitimate good reason to replace our NAs with 0s, which we can do with `values_fill()`:
 
-And so with this neat little trick, we've managed to bring the dataset into a better looking structure making further manipulation easier. Of course, this is not all we can do with the data, there are a wide range of tools at our disposal to allow us not only to pick the variables we need but also to split the contents of them into new columns, with more on that in the following sections. 
+```{r, purl=purl_example_code ,eval=T}
 
----
+# filling in NAs with 0
+wide_annual_offences <- annual_offences %>%
+  tidyr::pivot_wider(
+    names_from = 'year',
+    values_from = 'count',
+    names_prefix = 'count_',
+    values_fill = 0
+  )
+head(wide_annual_offences)
+```
 
-### Working with similar columns 
+Once our table is in wide format, and clean, we can easily do transformations like this. Here we use `dplyr` functions to create a new column that adds up yearly totals across each column that has count data.
 
-In some circumstances pivoting can also be used to construct a more robust for ease of use in later stages of the analysis. In these cases the dataset contains variables that are similar in their structure with the aim being to gather them in to one, cleaner, format. Consider the `anscombe` dataset that comes with basic R as shown below:
+```{r, purl=purl_example_code ,eval=T}
 
-```{r, purl=purl_example_code, eval=T}
-#dataset
-anscombe
+#starting by mapping a single month 
+wide_annual_offences_with_totals <- wide_annual_offences %>%
+  dplyr::mutate(
+    count_2016_2020 =
+      rowSums(dplyr::across(dplyr::starts_with("count"))))
 ```
 
-As it is apparent, the basis for each variable is the same (x1,x2, y1,y2 etc) and so it would make sense to simply create pair `x` and `y` and depict their corresponding identifier `1,2,3,4` in a separate variable. 
 
----
+The final and most advanced thing we will do with `pivot_wider()` is to pass it an auxiliary function to transform the values that it places in its new columns.
 
-In such cases, `pivot_longer` offers an elegant solution to the problem by automatically selecting the common patterns and using them as the new variables.
+Here we are passing an anonymous function which itself calls the `round()` function to round our counts. The `digits` argument tells `round()` to round one place before the decimal.
```

Comment:
The second sentence could instead be something like "Setting the `digits` argument of `round()` to -1 means the values get rounded to the nearest 10, rather than the default behaviour of rounding to the nearest whole number."